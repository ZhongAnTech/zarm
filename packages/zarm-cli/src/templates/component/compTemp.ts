import { paramCase } from 'change-case';

export default (compName) => `import React, { PureComponent } from 'react';
import classnames from 'classnames';

export interface ${compName}Props {
  prefixCls?: string;
  className?: string;
}

export interface ${compName}States {

}

export default class ${compName} extends PureComponent<${compName}Props, ${compName}States> {
  static displayName = '${compName}';

  static defaultProps = {
    prefixCls: 'za-${paramCase(compName)}',
  };

  render() {
    const {
      prefixCls,
      className,
      children,
    } = this.props;

    const cls = classnames(prefixCls, className);

    return (
      <div className={cls}>
        {children}
      </div>
    );
  }
}
`;
