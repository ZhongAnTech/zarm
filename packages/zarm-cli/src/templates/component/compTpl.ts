import { paramCase } from 'change-case';

export default (compName) => `import * as React from 'react';
import classnames from 'classnames';
import { ConfigContext } from '../n-config-provider';
import type { Base${compName}Props } from './interface';

export interface ${compName}Props extends Base${compName}Props {
  className?: string;
}

const ${compName} = React.forwardRef<unknown, ${compName}Props>((props, ref) => {
  const { className, children, ...restProps } = props;

  const compRef = (ref as any) || React.createRef<HTMLDivElement>();

  const { prefixCls: globalPrefixCls } = React.useContext(ConfigContext);
  const prefixCls = \`\${globalPrefixCls}-${paramCase(compName)}\`;
  const cls = classnames(prefixCls, className);

  return (
    <div ref={compRef} className={cls} {...restProps}>
      {children}
    </div>
  );
});

${compName}.displayName = '${compName}';

${compName}.defaultProps = {};

export default ${compName};

`;
