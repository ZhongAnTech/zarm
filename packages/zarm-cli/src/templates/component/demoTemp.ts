export default (compName) => `# ${compName} 组件名



## 基本用法
\`\`\`jsx
import { ${compName} } from 'zarm';

ReactDOM.render(
  <>
    <${compName} />
  </>
, mountNode);
\`\`\`



## 用法二
\`\`\`jsx
import { ${compName} } from 'zarm';

class Demo extends React.Component {
  state = {
  }

  render() {
    return <${compName} />;
  }
}

ReactDOM.render(<Demo />, mountNode);
\`\`\`



## API

| 属性 | 类型 | 默认值 | 说明 |
| :--- | :--- | :--- | :--- |
`;
