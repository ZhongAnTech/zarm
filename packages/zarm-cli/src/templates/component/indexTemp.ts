export default (compName) => `import ${compName} from './${compName}';

export default ${compName};
`;
