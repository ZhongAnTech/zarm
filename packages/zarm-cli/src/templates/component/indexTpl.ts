export default (compName) => `import ${compName} from './${compName}';

export type { ${compName}Props } from './${compName}';

export default ${compName};
`;
