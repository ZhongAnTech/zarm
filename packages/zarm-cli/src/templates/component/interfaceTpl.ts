export default (compName) => `export interface Base${compName}Props {

}

`;
