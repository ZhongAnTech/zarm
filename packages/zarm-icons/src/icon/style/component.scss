@import 'mixins';

@include b(icon) {
  display: inline-block;
  color: inherit;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &[type=search] {
    -webkit-appearance: initial !important;
  }

  /* icon themes */
  @include m(primary) {
    @include icon-theme(var(--theme-primary));
  }
  @include m(success) {
    @include icon-theme(var(--theme-success));
  }
  @include m(warning) {
    @include icon-theme(var(--theme-warning));
  }
  @include m(danger) {
    @include icon-theme(var(--theme-danger));
  }

  /* icon sizes */
  @include m(sm) {
    @include icon-size(var(--icon-font-size-sm));
  }
  @include m(md) {
    @include icon-size(var(--icon-font-size-md));
  }
  @include m(lg) {
    @include icon-size(var(--icon-font-size-lg));
  }
}
