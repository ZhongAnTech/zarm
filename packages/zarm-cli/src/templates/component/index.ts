import compTemp from './compTemp';
import indexTemp from './indexTemp';
import demoTemp from './demoTemp';

export default {
  compTemp,
  indexTemp,
  demoTemp,
};
