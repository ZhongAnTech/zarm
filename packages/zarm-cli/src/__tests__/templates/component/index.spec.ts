import component from '../../../templates/component';

describe('templates', () => {
  describe('component', () => {
    it('should create component template', () => {
      const template = component.compTemp('MyTestComponent');
      expect(template).toMatchSnapshot();
    });
    it('should create demo template', () => {
      const template = component.demoTemp('MyTestComponent');
      expect(template).toMatchSnapshot();
    });
    it('should create index template', () => {
      const template = component.indexTemp('MyTestComponent');
      expect(template).toMatchSnapshot();
    });
  });
});
