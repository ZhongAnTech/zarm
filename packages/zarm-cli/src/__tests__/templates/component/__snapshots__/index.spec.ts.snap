// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates component should create component template 1`] = `
"import React, { PureComponent } from 'react';
import classnames from 'classnames';

export interface MyTestComponentProps {
  prefixCls?: string;
  className?: string;
}

export interface MyTestComponentStates {

}

export default class MyTestComponent extends PureComponent<MyTestComponentProps, MyTestComponentStates> {
  static displayName = 'MyTestComponent';

  static defaultProps = {
    prefixCls: 'za-my-test-component',
  };

  render() {
    const {
      prefixCls,
      className,
      children,
    } = this.props;

    const cls = classnames(prefixCls, className);

    return (
      <div className={cls}>
        {children}
      </div>
    );
  }
}
"
`;

exports[`templates component should create demo template 1`] = `
"# MyTestComponent 组件名



## 基本用法
\`\`\`jsx
import { MyTestComponent } from 'zarm';

ReactDOM.render(
  <>
    <MyTestComponent />
  </>
, mountNode);
\`\`\`



## 用法二
\`\`\`jsx
import { MyTestComponent } from 'zarm';

class Demo extends React.Component {
  state = {
  }

  render() {
    return <MyTestComponent />;
  }
}

ReactDOM.render(<Demo />, mountNode);
\`\`\`



## API

| 属性 | 类型 | 默认值 | 说明 |
| :--- | :--- | :--- | :--- |
"
`;

exports[`templates component should create index template 1`] = `
"import MyTestComponent from './MyTestComponent';

export default MyTestComponent;
"
`;
